experimental/framework/fluid-static/src/FluidStatic.ts

```diff
@@ -4,64 +4,129 @@
  */
 
 import { getContainer, IGetContainerService } from "@fluid-experimental/get-container";
-import { getObjectWithIdFromContainer } from "@fluidframework/aqueduct";
+import { DataObject, getObjectWithIdFromContainer } from "@fluidframework/aqueduct";
 import { IContainer } from "@fluidframework/container-definitions";
-import { IFluidDataStoreFactory, NamedFluidDataStoreRegistryEntry } from "@fluidframework/runtime-definitions";
-import { DOProviderContainerRuntimeFactory } from "./containerCode";
+import { NamedFluidDataStoreRegistryEntry } from "@fluidframework/runtime-definitions";
+import { DOProviderContainerRuntimeFactory, IFluidStaticDataObjectClass } from "./containerCode";
 
-export interface IFluidStaticDataObjectClass {
-    readonly factory: IFluidDataStoreFactory;
+export interface FluidContainerConfig {
+    id: string,
+    dataObjects: IFluidStaticDataObjectClass[],
+}
+
+export interface FluidCreateContainerConfig extends FluidContainerConfig {
+    /**
+     * initial DataObjects will be created when the Container is first created.
+     */
+    initialDataObjects?: [string, IFluidStaticDataObjectClass][]
 }
 
 export class FluidContainer {
-    constructor(private readonly container: IContainer, public readonly createNew: boolean) { }
+    private readonly types: Set<string>;
+    constructor(
+        private readonly container: IContainer,
+        namedRegistryEntries: NamedFluidDataStoreRegistryEntry[],
+        public readonly createNew: boolean) {
+        this.types = new Set();
+        namedRegistryEntries.forEach((value: NamedFluidDataStoreRegistryEntry) => {
+            const type = value[0];
+            if (this.types.has(type)) {
+                throw new Error(`Multiple DataObjects share the same type identifier ${value}`);
+            }
+            this.types.add(type);
+        });
+    }
 
-    public async createDataObject<T = any>(dataObjectClass: IFluidStaticDataObjectClass, id: string) {
+    public async createDataObject<T extends DataObject>(
+        dataObjectClass: IFluidStaticDataObjectClass, id: string): Promise<T> {
         const type = dataObjectClass.factory.type;
+        // This is a runtime check to ensure the developer doesn't try to create something they have not defined.
+        if (!this.types.has(type)) {
+            throw new Error("Trying to create a DataObject that was not defined in the Fluid.init");
```

Comment:
What are you referring to with "Fluid.init?"

Reply:
Oh, I see now.

```ts
const service = new TinyliciousService();
Fluid.init(service);
```

I'm sill confused though because the ewrror message implies I didn't include something in Fluid.init, but DOs don't appear to be defined in the call to Fluid.init. What am I missing?

Reply:
You're right. I originally had the DataObject definition happening in the `Fluid.init` which is where this comment came from. The strong binding between the init and the DataObjects seemed artificial. I decided that defining the Container config was better but yea didn't update this comment. Nice catch